lib/proxy/http_downloads.rb

```diff
@@ -1,29 +1,27 @@
 require 'proxy/file_lock'
 
 module Proxy
-  class HttpDownloads
-    class << self
-      def start_download(src, dst)
-        lock =  Proxy::FileLock.try_locking(dst)
-        unless lock.nil?
-          HttpDownload.new(src, dst, lock)
-          return true
-        end
-        false
-      end
-    end
-  end
-
   class HttpDownload < Proxy::Util::CommandTask
     include Util
 
-    def initialize(src, dst, lock)
-      super(command(src, dst)) {  Proxy::FileLock.unlock(lock) }
+    def initialize(src, dst, lock_dst = true)
```

Comment:
I can't think of any cases when we don't want to ensure that only a single download is in progress -- it results in corrupted files. What was the reason for adding of this flag?